### PR TITLE
Add int16 support for SQUARED_DIFFERENCE

### PR DESCRIPTION
**Pull request by rewu93**
BUG=http://b/269352046

Add INT16 support for SQUARED_DIFFERENCE by templating the kernel implementation. Reuse int8 test settings to test the int16 implementation as well

### CODE REVIEW

File: tensorflow/lite/micro/kernels/squared_difference.cc [hunk 1]
```diff
@@ -124,6 +124,56 @@ TfLiteStatus SquaredDifferencePrepare(TfLiteContext* context,
         std::numeric_limits<int8_t>::min();
     data->arithmetic_params.quantized_activation_max =
         std::numeric_limits<int8_t>::max();
+  } else if (input1->type == kTfLiteInt16) {
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/squared_difference.cc:127]: only the TF_LITE_ENSUREs are different between int8 and int16, right? And the last QuantizeMultipler.

Can we share the rest of the code? We will still need an if for quantized type vs float, right?

If it helps you can break up the logic into a PrepareQuantized that only gets called if input type is not a float.

File: tensorflow/lite/micro/kernels/squared_difference.cc [hunk 2]
```diff
@@ -124,6 +124,56 @@ TfLiteStatus SquaredDifferencePrepare(TfLiteContext* context,
...
+    const auto& input1_quantization_params = input1->params;
+    const auto& input2_quantization_params = input2->params;
+    const auto& output_quantization_params = output->params;
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/squared_difference.cc:130]: let's not use auto here since the types are not apparent from the immediate context.

**Review by advaitjain — CHANGES_REQUESTED** [review]

File: tensorflow/lite/micro/kernels/squared_difference.cc (hunk 1 above)
Comment (rewu93) [reply tensorflow/lite/micro/kernels/squared_difference.cc:133]: Done.

File: tensorflow/lite/micro/kernels/squared_difference.cc (hunk 2 above)
Comment (rewu93) [reply tensorflow/lite/micro/kernels/squared_difference.cc]: Done.

**Review by advaitjain — APPROVED** [review]

**Pull request merged**